distrobuilder/main.go

```diff
@@ -258,6 +258,15 @@ func (c *cmdGlobal) preRunBuild(cmd *cobra.Command, args []string) error {
 	// Unmount everything and exit the chroot
 	defer exitChroot()
 
+	// If we're running build-dir, only process sections which are meant for both
+	// containers and VMs.
+	imageTargets := shared.ImageTargetAll
+
+	// If we call build-lxc or build-lxd, also process container-only sections.
+	if cmd.CalledAs() != "build-dir" {
+		imageTargets |= shared.ImageTargetContainer
+	}
```

Comment:
So presumably you'll have logic in the VM branch so that when running `pack-lxd --vm` or `build-lxd --vm` that we set `ImageTargetVirtualMachine` and don't set `ImageTargetContainer` in that case?

Reply:
Exactly.